### ISSUE TITLE
Get rebooted after creating the one big partition!

### ISSUE THREAD

**Issue by Backspin7792** [raspiblitz/raspiblitz#4231]
**Describe the bug**
I get rebooted after creating the big one partition

**To Reproduce**
Start fresh new installation on Odroid m1 SD (os) and ext4 nvme 

**Debug file**
https://l.termbin.com/a58y

**Comment by Backspin7792** [comment]
Update: i tryed the installation again. This time i got stucked at the provision process:

https: //l.termbin.com/36ri

**Comment by Backspin7792** [comment]
I have tryed to reinstall several times.. I got the same problem:

![IMG_20231028_133140](https://github.com/raspiblitz/raspiblitz/assets/127111218/c6c2c547-6729-4181-a3f3-b092865d72ca)

But 2 times i managed to advance till here and get stucked:

![IMG_20231028_133239](https://github.com/raspiblitz/raspiblitz/assets/127111218/a6562a1f-72f0-4c1e-bd16-1ad675f9496a)

https://l.termbin.com/v9jr

**Comment by openoms** [comment]
A reboot should not be triggered there so I presume it has something to do with the hardware. 

Try to use the disk with another OS, maybe just your base to see if you can mount and format it without issues,

There could be a problem with the power supply, nvme adapter or something else specific to the Odroid  M1. Might eb worth to search around in the Odroid forums as well.

**Comment by Backspin7792** [comment]
I had this same hardware running smoothly with another implementation for several months and no issue. The only thing i changed is the SD card i added where i placed the OS. The power adapter is brand new and tested for some days. Don't think it's hardware problem..

**Comment by openoms** [comment]
> I had this same hardware running smoothly with another implementation for several months and no issue. The only thing i changed is the SD card i added where i placed the OS. The power adapter is brand new and tested for some days. Don't think it's hardware problem..

Did you use the disk as well with the previous setup? 

The way to test this would be to follow the commands being run and execute them manually to see if any of them causes a reboot.

I can't reproduce it myself as I don;t have the hardware.

**Comment by Backspin7792** [comment]
> > I had this same hardware running smoothly with another implementation for several months and no issue. The only thing i changed is the SD card i added where i placed the OS. The power adapter is brand new and tested for some days. Don't think it's hardware problem..
> 
> Did you use the disk as well with the previous setup?
> 
> The way to test this would be to follow the commands being run and execute them manually to see if any of them causes a reboot.

Yes i did, there's a guide i can follow to test commands manually? I'm not so technically skilled to do it by myself :)

**Comment by openoms** [comment]
There is no guide, it would involve reading the raspiblitz scripts.

For a manual setup I would recommend following the steps on raspibolt.org,  the resulting system is largely similar.

**Comment by Backspin7792** [comment]
I've already performed raspibolt on another node (same set up of this one), i would like to try and run raspiblizt explore this implementation's gui and upgradeability..

**Comment by openoms** [comment]
If you understand what it means (deletes the disk contents) you can try to run this outside of the script with sudo:

https://github.com/raspiblitz/raspiblitz/blob/bf9aaa9076246ec1f517831bf4d4444e7086a67b/home.admin/config.scripts/blitz.datadrive.sh#L729-L731


get the value for $hdd variable with (could help if you post the output of it):
```
config.scripts/blitz.datadrive.sh status
```

set it before running the above line with:
```
hdd=nvme0n1
```

Can also try wiping the disk mnaually before starting again or running the above with:
```
sudo fdisk --delete /dev/${hdd}
sudo wipefs -a /dev/${hdd}
```

**Comment by Backspin7792** [comment]
As sudo user: i setted the var and deleted the partition, used wipesf then tryed

>  sudo >&2 echo "# Creating the one big partition" 
>  sudo  parted -s /dev/${hdd} mkpart primary ext4 1024KiB 100% 1>&2  

but here i've got "Error: /dev/nvme0n1: unrecognized disc label"

so i ran:  sudo config.scripts/blitz.datadrive.sh status , this is the result:

# RASPIBLITZ DATA DRIVE Status 

# BASICS 
isMounted=1 
isBTRFS=0 
hddRaspiData=1 
hddRaspiVersion='1.10.0' 
isSMART=0 
isSSD=1 
datadisk='nvme0n1' 
datapartition='nvme0n1p1' 
hddCandidate='nvme0n1' 
hddPartitionCandidate='nvme0n1p1' 
hddBlocksBitcoin=0 
hddBytes=2000397795328 
hddGigaBytes=1863 
ddUsedInfo='100M (1%)' 
hddDataFreeBytes=1867784695808 
hddDataFreeKB=1824008492 
hddDataFreeGB=1739 
hddAdapterUSB='' 
hddAdapterUSAP=0 

# RAID 
isRaid=0 
raidCandidate[0]='mtdblock0 0 GB mtdblock0' 
raidCandidate[1]='mtdblock1 0 GB mtdblock1' 
raidCandidate[2]='mtdblock2 0 GB mtdblock2' 
raidCandidate[3]='mtdblock3 0 GB mtdblock3' 
raidCandidates=4 

# SWAP 
isSwapExternal=0

**Comment by openoms** [comment]
Looks much different from the RPi or amd64 setup.

use the command:
`lsblk`
to list the devices. Please share that too.

**Comment by Backspin7792** [comment]
![Uploading IMG_20231031_155325.jpg…]()

**Comment by Backspin7792** [comment]
![IMG_20231031_155325](https://github.com/raspiblitz/raspiblitz/assets/127111218/3f6f5bd7-969f-4bbe-adbc-4f2e1d811296)

**Comment by openoms** [comment]
that shows that your SDcard (or MMC) has a lot of partitions, but that shouldn't be a problem.

Try running the parted command with:
`hdd=nvme0n1p1`

That shouldn't cause a restart.

**Comment by Backspin7792** [comment]
this didn't work cause unrecognized disc (nvme0n1 nor nvme0n1p1) label like above, i had to: 

sudo parted /dev/nvme0n1 mklabel gpt 
sudo parted -s /dev/nvme0n1 mkpart primary ext4 1024KiB 100% 1>&2  
sudo reboot

at this point i tryed to acces admin but i got a window tells me to start with a fresh sd card image, i logged the main user (which i installed the os) and i ran sudo ./build_sdcard.sh , this is the result:

sudo su - admin 

Starting SSH user interface ... (please wait) 
setupPhase() 
state() FAIL: /home/admin/raspiblitz.info does not exist or missing state. 
Check logs & bootstrap.service for errors and report to devs.

**Comment by openoms** [comment]
Running the `build_sdcard.sh` repeatedly is not an intended way to reset the system.

What you could do to be able to start clean is 
* set up your base image new (needs to be Debian or Ubuntu - please share what that is)
* run the `build_sdcard.sh` 
* once it has finished switch off and make an image of your SDcard/eMMC so you can flash it again to restore this initial state

Unfortunately this so far did not bring us closer why the system reboots, but it is strange to me why it does not even happen all the time.

**Comment by Backspin7792** [comment]
oh, i didn't know:

i use ubuntu 20.04 cause is the best stable distro on Odroid M1.

>Unfortunately this so far did not bring us closer why the system reboots, but it is strange to me why it does not even happen > all the time.

Yeah this is crazy. I'm here to help if you wanna debug this one i will take this node for this purpose. Next step i will reinstall ubuntu, format and partition the nvme disk and make an image of it.. so i'ill retry to build the raspiblitz and update this post. Tell me if you think i have to do something different and have a nice halloween 🎃

**Comment by Backspin7792** [comment]
> Running the `build_sdcard.sh` repeatedly is not an intended way to reset the system.
> 
> What you could do to be able to start clean is
> 
> * set up your base image new (needs to be Debian or Ubuntu - please share what that is)
> * run the `build_sdcard.sh`
> * once it has finished switch off and make an image of your SDcard/eMMC so you can flash it again to restore this initial state
> 
> Unfortunately this so far did not bring us closer why the system reboots, but it is strange to me why it does not even happen all the time.

I did all the process from the start but got no lucky as i said before, have you further suggestion to solve/debug the problem?

**Comment by freerko** [comment]
as mentioned in https://github.com/raspiblitz/raspiblitz/pull/4238, I got this issue as well with 2 old hdd disks but as well with my new 2TB Kingston XS1000 SSD. It doesn't matter if I format the disks on a Linux system before as ext4 or FAT or on a Windows system as NTFS, the RaspiBlitz reboots either right before "Creating the one big partition" or right after: 

```
Running Format: (ext4) (sda)
# DATA DRIVE - FORMATTING to EXT4 layout (old)
# Unmounting all partitions of this device
# Wiping all partitions (sfdisk/wipefs)

Broadcast message from root@raspberrypi (Thu 2023-11-23 15:19:36 GMT):

The system will reboot now!


Broadcast message from root@raspberrypi (Thu 2023-11-23 15:19:36 GMT):

The system will reboot now!

Connection to 192.168.178.28 closed by remote host.
Connection to 192.168.178.28 closed.
```
With "sudo fdisk -l" on the RaspiBlitz 0.10.0 (Bookworm) I see the SSD is recognized as /dev/sda. After the reboot the partitions are gone and RaspiBlitz doesn't recognize the disk anymore and hangs in the "Waiting for HDD/SSD" menu, I have to create a ext4/FAT/NTFS partition on another system again.

```
admin@192.168.178.28:~ ₿ lsblk
NAME        MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
sda           8:0    0   1.8T  0 disk 
mmcblk0     179:0    0 119.4G  0 disk 
├─mmcblk0p1 179:1    0   512M  0 part /boot/firmware
└─mmcblk0p2 179:2    0 118.9G  0 part /
```
The blitz.datadrive.sh script knows nothing about a disk apparently:

```
admin@192.168.178.28:~ ₿ sudo /home/admin/config.scripts/blitz.datadrive.sh status
# RASPIBLITZ DATA DRIVE Status

# BASICS
isMounted=0
isBTRFS=0

# SETUP INFO
isSMART=0
isSSD=0
hddCandidate=''
hddBytes=0
hddGigaBytes=0
hddPartitionCandidate=''

# RAID
isRaid=0
raidCandidates=0

# SWAP
isSwapExternal=0

admin@192.168.178.28:~ ₿
```
How can I debug this further?

**Comment by rootzoll** [comment]
seems simlar to #4290

try if the following RaspberryPi image (Download under Artifacts) fixes this:
https://github.com/raspiblitz/raspiblitz/actions/runs/7201966440

**Comment by Backspin7792** [comment]
> seems simlar to #4290
> 
> try if the following RaspberryPi image (Download under Artifacts) fixes this: https://github.com/raspiblitz/raspiblitz/actions/runs/7201966440

i flashed the image on the sd card and rebooted the odroid m1 but it didn't recognize any os. Any clue?

**Comment by rootzoll** [comment]
@Backspin7792 the image is just for RaspberryPi odroid might need a special image ... @openoms is there a odroid build?

**Comment by Backspin7792** [comment]
> @Backspin7792 the image is just for RaspberryPi odroid might need a special image ... @openoms is there a odroid build?

there's a way to get the right build_sdcard.sh so i can test it without an image?

**Comment by openoms** [comment]
@Backspin7792 you can run the build_sdcard.sh on the Odroid once you booted up and connected with a Debian or Ubuntu base image or can try Armbian as described in: https://github.com/raspiblitz/raspiblitz/tree/dev/alternative.platforms#armbian

**Comment by Backspin7792** [comment]
> @Backspin7792 you can run the build_sdcard.sh on the Odroid once you booted up and connected with a Debian or Ubuntu base image or can try Armbian as described in: https://github.com/raspiblitz/raspiblitz/tree/dev/alternative.platforms#armbian

yeah i did it several times but i got the hd error described above.. i'm wondering if the build_sdcard.sh is updated and can solve the issue, or i have to modify something.. ty
